tasks/main.yml

```diff
@@ -4,5 +4,77 @@
     that: agent_type in ['monitoring', 'logging', 'ops-agent']
     msg: "Received invalid agent type: '{{ agent_type }}'. The Cloud Ops Ansible role supports the following agents: 'monitoring', 'logging' and 'ops-agent'."
 
-- name: Call agent-specific playbook
-  include_tasks: "{{ agent_type }}.yml"
+- name: Validate package_state
+  assert:
+    that: package_state in ['present', 'absent']
+    msg: "Received invalid package state: '{{ package_state }}'. The Cloud Ops Ansible role supports the following package states: 'present' and 'absent'."
+
+- when: agent_type == 'ops-agent'
+  block:
+    - name: Ensure no additional config directory was specified
+      assert:
+        that: not additional_config_dir
+        msg: 'The ops agent does not support additional configurations.'
```

Comment:
How about `The ops agent does not support additional configurations. additional_config_dir must be empty when agent_type is ops-agent.` Just in case users are not familiar with the configuration enough to immediately figure out what action items to take.

Side notes: It may make sense to have Jimmy take a quick look at all the validation error messages that are user visible to check the phrasing from tech writing side.

Reply:
Sgtm, created go/cloud-ops-ansible-user-messages for Jimmy to review.